contracts/Depository.sol

```diff
@@ -326,16 +381,34 @@ contract Depository is IErrorsTokenomics {
 
         // Decrease the supply for the amount of payout
         supply -= payout;
+        // Adjust payout and set supply to zero if supply drops below the min defined value
+        if (supply < minOLASLeftoverAmount) {
+            payout += supply;
+            supply = 0;
+        }
         product.supply = uint96(supply);
+        product.payout += uint96(payout);
+
+        // Create and mint a new bond
+        bondId = totalSupply;
+        // Safe mint is needed since contracts can create bonds as well
+        _safeMint(msg.sender, bondId);
```

Comment:
audit notes: re-check reentrancy in internal audit

Reply:
service Ids indexing must start from zero due to possible mapping conflicts. There is no such thing for bonds.